src/common/url-decode.decorator.ts

```diff
@@ -0,0 +1,11 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+
+/**
+ * Decodes URL encoded request url
+ */
+export const UrlDecoded = createParamDecorator(
```

Comment:
"request url" is ambiguous between just the "original URL" that we are using or the whole URL with the domain, so better to be more specific
```suggestion
/**
 * Decodes URL encoded "originalUrl"
 * https://expressjs.com/en/api.html
 */
export const OriginalUrlDecoded = createParamDecorator(
```